src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3518,6 +3516,13 @@ async def __common_validation(self, verrors, schema, data, mode, parent=None, cu
                     f'{schema}.special_small_block_size',
                     'This field can be "INHERIT", 0 or multiple of 512, up to 1048576'
                 )
+            if rs := data.get('recordsize'):
+                choices = await self.middleware.call('pool.dataset.recordsize_choices')
+                if rs not in choices:
+                    verrors.add(
+                        f'{schema}.recordsize',
+                        '{rs!r} is an invalid recordsize. Valid choices are {", ".join(choices)}'
```

Comment:
We never do this elsewhere `Valid choices are {", ".join(choices)}` and this list can be pretty big

Reply:
Okay, but why does that matter? How do you propose we give the consumer the valid available choices?

Reply:
Via `pool.dataset.recordsize_choices` which should be a documented public method

Reply:
So making them read the API documentation when they given an invalid recordsize in the webUI is what you're suggesting?

Reply:
Furthermore, if you read the code you'll see that we're doing _exactly_ what I'm doing just a few lines above
```
3515                 verrors.add(
3516                     f'{schema}.special_small_block_size',
3517                     'This field can be "INHERIT", 0 or multiple of 512, up to 1048576'
3518                 )
```